lib/src/layer/label.dart

```diff
@@ -1,26 +1,30 @@
-import 'dart:math';
+import 'dart:math' as math;
 
+import 'package:collection/collection.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_map/plugin_api.dart';
+import 'package:polylabel/polylabel.dart';
 
 class Label {
   static void paintText(
     Canvas canvas,
     List<Offset> points,
     String? labelText,
-    TextStyle? labelStyle,
-  ) {
-    var maxDx = 0.0;
-    var minDx = double.infinity;
-    var dx = points.map((e) => e.dx).toList().fold<double>(0.0,
-            (previousValue, element) {
-          maxDx = max(maxDx, element);
-          minDx = min(minDx, element);
-          return previousValue + element;
-        }) /
-        points.length;
-    var dy = points.map((e) => e.dy).toList().fold<double>(
-            0.0, (previousValue, element) => previousValue + element) /
-        points.length;
+    TextStyle? labelStyle, {
+    PolygonLabelPlacement labelPlacement = PolygonLabelPlacement.centroid,
```

Comment:
I think we should change the default to this new method, but that's just my opinion.